### ISSUE TITLE
Infinite recursion on `startNewSessionIfNecessary()`

### ISSUE THREAD

**Issue by ivan-r** [segment-integrations/analytics-swift-amplitude#22]
With the latest changes introduced in v1.4.0 an infinite recursion is occuring with `startNewSessionIfNecessary()`.

If `sessionId` == -1
```
public func execute<T: RawEvent>(event: T?) -> T?
```
will call
```
startNewSessionIfNecessary()
```
that will call 
```
endSession()
```
that will call 
```
analytics?.track(name: Constants.ampSessionEndEvent)
```
that will call 
```
public func execute<T: RawEvent>(event: T?) -> T?
```
Staring the loop again, until the app crashes

**Comment by ivan-r** [comment]
In addition, there seem to be no place where `sessionID` is being given a value.

**Comment by bsneed** [comment]
Thanks for reporting this @ivan-r, this is fixed now in 1.4.1.
